Applications/TransferLearning/Draw_Classification/jni/main.cpp

```diff
@@ -62,15 +68,122 @@
 
 /** Vectors containing the training data */
-std::vector<std::vector<float>> inputVector, labelVector;
+float inputVector[EPOCH_SIZE][INPUT_SIZE];
+float labelVector[EPOCH_SIZE][LABEL_SIZE];
+
+#if defined(__TIZEN__)
+int getInputFeature(ml_single_h single, const char *test_file_path,
+                    float *feature_input) {
+  int status = ML_ERROR_NONE;
+  ml_tensors_info_h in_res;
+  ml_tensors_data_h input = NULL, output;
+  void *raw_data;
+  size_t data_size;
+  int inputDim[MAX_DIM] = {1, 1, 1, 1};
+  uint8_t *in = NULL;
+
+  /* input tensor in filter */
+  status = ml_single_get_input_info(single, &in_res);
+  if (status != ML_ERROR_NONE)
+    return status;
+
+  /* generate dummy data */
+  status = ml_tensors_data_create(in_res, &input);
+  if (status != ML_ERROR_NONE)
+    goto fail_in_info;
+
+  status = ml_tensors_data_get_tensor_data(input, 0, &raw_data, &data_size);
+  if (status != ML_ERROR_NONE)
+    goto fail_in_data;
+
+  in = tflite::label_image::read_bmp(test_file_path, inputDim, inputDim + 1,
+                                     inputDim + 2);
+
+  for (size_t l = 0; l < data_size / sizeof(float); l++) {
+    ((float *)raw_data)[l] = ((float)(in[l]) - 127.5f) / 127.5f;
+  }
+  delete[] in;
+
+  status = ml_single_invoke(single, input, &output);
+  if (status != ML_ERROR_NONE)
+    goto fail_in_data;
+
+  status = ml_tensors_data_get_tensor_data(output, 0, &raw_data, &data_size);
+  if (status != ML_ERROR_NONE)
+    goto fail_out_data;
+
+  memcpy(feature_input, raw_data, INPUT_SIZE * sizeof(float));
+
+fail_out_data:
+  ml_tensors_data_destroy(output);
+
+fail_in_data:
+  ml_tensors_data_destroy(input);
+
+fail_in_info:
+  ml_tensors_info_destroy(in_res);
+
+  return status;
+}
+
+int setupSingleModel(const char *data_path, ml_single_h *single) {
+  int status;
+
+  char *test_model = NULL;
+  status = asprintf(&test_model, "%s/%s", data_path,
+                    "ssd_mobilenet_v2_coco_feature.tflite");
+  if (status < 0 || test_model == NULL)
+    return -errno;
+
+  status = ml_single_open(single, test_model, NULL, NULL,
+                          ML_NNFW_TYPE_TENSORFLOW_LITE, ML_NNFW_HW_ANY);
+  free(test_model);
+
+  return status;
+}
+
+int extractFeatures(const char *data_path, float input_data[][INPUT_SIZE],
+                    float label_data[][LABEL_SIZE]) {
+  ml_single_h single;
+  unsigned int count = 0;
+  int status;
+
+  status = setupSingleModel(data_path, &single);
+  if (status != ML_ERROR_NONE)
+    return status;
+
+  for (int i = 0; i < LABEL_SIZE; i++) {
+    for (int j = 0; j < NUM_DATA_PER_LABEL; j++) {
+
+      char *test_file_path = NULL;
+      status = asprintf(&test_file_path, "%s/%s/%s%d.bmp", data_path,
+                        label_names[i], label_names[i], j + 1);
+      if (status < 0 || test_file_path == NULL) {
+        status = -errno;
+        goto fail_exit;
+      }
+
+      count = i * NUM_DATA_PER_LABEL + j;
+      memset(label_data[count], 0, LABEL_SIZE * sizeof(float));
```

Comment:
It seems that some architecture might not be using `all bit 0` as representing 0.0f. 
Check this out
https://stackoverflow.com/a/37555983

Reply:
done.